src/connectdlg.cpp

```diff
@@ -731,6 +740,12 @@ void CConnectDlg::OnConnectClicked()
     done ( QDialog::Accepted );
 }
 
+void CConnectDlg::OnDeleteServerAddrClicked()
+{
+    pSettings->vstrIPAddress.erase ( pSettings->vstrIPAddress.begin() + cbxServerAddr->currentIndex() );
```

Comment:
Having built on my pi, clicking the Delete button crashes the app, reliably. If I comment out this line and rebuild, it doesn't crash (but of course doesn't delete). Will look more closely.

Reply:
What Qt version are you building with?  I tested with Qt6 only.

Reply:
5.15.2 on the Pi. I don't think it's the Qt version. My suspicion is that `vstrIPAddress.erase` is changing the size of `vstrIPAddress`, which the rest of the code doesn't expect. Haven't proved or disproved it yet. I see there are no calls to an `erase()` method on a vector anywhere else in the code.

Reply:
Mmmm, reading the C++ definition now, I don't see why I used that operation.

Reply:
Just about to test an alternative.

Reply:
OK, replaced with a different approach which seems to work.